components/automate-ui/src/app/helpers/auth/regex.ts

```diff
@@ -19,6 +19,13 @@ export class Regex {
     // Allows no special characters except hyphen and underscore.
     NO_WILDCARD_ALLOW_HYPHEN: /^[0-9a-zA-Z-_]+$/,
 
+    // Only allows words or numbers or hyphen or underscore or dot also combined
+    // Note: the first character should not be dot(.).
+    // Legal Values: _, -, chef, _state, -chef, test-23-23.com-1.test.com etc.
+    // Illegal Values: .chef, *chef, ., c*h*e*f, **, * chef
+    // Allows no special characters except hyphen, underscore and dot(.).
+    NO_WILDCARD_ALLOW_HYPHEN_AND_DOT: /^[0-9a-zA-Z-_]+[0-9a-zA-Z-_.]*$/,
```

Comment:
/^[0-9a-zA-Z-_.]*$/  just this pattern works.

Reply:
@iamazzeez  the above pattern allows sending an empty search word and a single dot(.) as a search keyword which we should not allow.
<img width="208" alt="image" src="https://user-images.githubusercontent.com/44021874/192773063-3ffd5c89-7905-46f8-a060-f22fd24cc747.png">

Reply:
Got it.